pkg/kv/mem/store.go

```diff
@@ -66,14 +55,14 @@ func (s *Store) Get(_ context.Context, partitionKey, key []byte) (*kv.ValueWithP
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
-	combinedKey := combinedKey(partitionKey, key)
-	value, ok := s.m[combinedKey]
+	sKey := encodeKey(key)
+	value, ok := s.m[string(partitionKey)][sKey]
```

Comment:
What is the reason to allow plain strings for partition keys?
Why not encoding it too?

Reply:
Since we control the partition key, we can take advantage of that and assume it can be converted to string